voicevox_engine/setting/Setting.py

```diff
@@ -0,0 +1,12 @@
+from typing import Optional
+
+from pydantic import BaseModel, Field
+
+
+class Setting(BaseModel):
+    """
+    エンジンの設定情報
+    """
+
+    cors_policy_mode: str = Field(title="リソース共有ポリシー")
```

Comment:
strだと`run.py` に定義されている`CorsPolicyMode`以外の値も受け入れてしまうので、制限する方が良いかと思います。

Reply:
FastAPI周りの知識が必要なので、わからなければFIXMEコメントでも良いかもですね！

Reply:
できれば一つに纏めたいですね...
少し大変そうなので、同じくfixmeコメントでもいいかなと思います。
```suggestion
    # FIXME: run.pyのCorsPolicyModeと纏める
    cors_policy_mode: str = Field(title="リソース共有ポリシー")
```

Reply:
これに関しましては、昨日同じように一つにまとめたいと思いEnum型にしてみたりしていたのですが、yamlでの書き込みの際におかしな書き込みになってしまいできませんでした。pydantic側は対応していて正常に処理できたのですが、PyYAMLとの相性が悪い感じがします。シリアライズ化をすればいけなくもない気がするのですが、処理が複雑化してしまう可能性があるためやりませんでした。

Reply:
横から失礼します.
以下の2点を実装すれば対応できると思います:

1. Enum クラスで `str` を inherit

```python
from enum import Enum

class OptionEnum(str, Enum):
    OPTION_1 = "OPTION_1"
    OPTION_2 = "OPTION_2"
```

2. モデル設定で `use_enum_values` を指定する

```python
from pydantic import BaseModel

class Config(BaseModel):
    option: OptionEnum

    class Config:
        use_enum_values = True
````

https://docs.pydantic.dev/usage/model_config/#options

Reply:
おお、確かにできました。Configまで見ることができていませんでした。これを使って統合と、制限を進めていきます。ありがとうございます。